### PR TITLE
feat(kubernetes-etcd-backup): skip tls verify

### PR DESCRIPTION
**Pull request by in0rdr**
etcdctl can be run with the `--insecure-skip-tls-verify` to skip tls verification of the etcd endpoint.

This is useful in some deployments, for instance, when the etcd cluster is external to Kubernetes and the Kubernetes endpoint name (e.g., `etcd.kube-system.svc.cluster.local`) does not match the names in the certificates of the external etcd cluster.

# Checklist

* [x] This PR contains a description of the changes I'm making
* [x] I updated the version in Chart.yaml
  * bumped the minor version
* [x] I updated the changelog with an `artifacthub.io/changes` annotation in `Chart.yaml`, check the [example](docs/development.md#Changelog) in the documentation.
* [x] I updated applicable README.md files using  `pre-commit run`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

**Review by vmaillot — APPROVED** [review]

**Comment by 4censord** [comment]
Wouldn't it be better to just specify the correct hostname?
Or is that not an option in your case?

**Comment by 4censord** [comment]
Hmm, how does the kubelet resolve/verify etdc then?
Or does the kubelet just not verify as well?

**Comment by tongpu** [comment]
> This is useful in some deployments, for instance, when the etcd cluster is external to Kubernetes and the Kubernetes endpoint name (e.g., `etcd.kube-system.svc.cluster.local`) does not match the names in the certificates of the external etcd cluster.

AFAIK only the Kubernetes API server talks directly to etcd, so only they would need to have the correct hostname configured.

File: charts/kubernetes-etcd-backup/values.yaml [hunk 1]
```diff
@@ -22,7 +22,16 @@ backup:
...
+  # Example: etcd.kube-system.svc.cluster.local
   endpoint: "changeme"
+  # -- Skip server certificate verification
+  # Useful for scenarios where etcd nodes are external endpoints (access
+  # through etcd service in kube-system namespace) and have a different CN/SAN
+  # in the certificate . Otherwise, "failed to verify certificate: x509:
+  # certificate is valid for etcd-2, etc., not
+  # etcd.kube-system.svc.cluster.local"
+  insecureSkipTlsVerify: false
+
```
Comment (tongpu) [inline charts/kubernetes-etcd-backup/values.yaml:34]: Superfluous empty line

```suggestion
```

**Comment by eyenx** [comment]
> Hmm, how does the kubelet resolve/verify etdc then? Or does the kubelet just not verify as well?

The IPs are hardcoded in the Endpoints CR. The TLS skip-verify is only needed cause Nutanix's Implementation of ETCD Cluster uses a certificate that is only valid for 
* `hostname` (not resolvable within cluster)
* `hostname.cluster.local` (no idea why that would help)
* `*.cluster.local` (does not help as well)

**Comment by hairmare** [comment]
> The TLS skip-verify is only needed cause Nutanix's Implementation of ETCD

Should we also reflect this issue in upstream Nutanix so it gets unborked at some point?

**Comment by 4censord** [comment]
> ETCD Cluster uses a certificate that is only valid for
> `*.cluster.local` (does not help as well)

Now I'm even more confused, because shouldn't that be valid for the internal service `etcd.kube-system.svc.cluster.local`?

**Comment by tongpu** [comment]
> > ETCD Cluster uses a certificate that is only valid for
> > 
> > * `*.cluster.local` (does not help as well)
> 
> Now I'm even more confused, because shouldn't that be valid for the internal service `etcd.kube-system.svc.cluster.local`?

No, because the wildcard is only valid for a single subdomain, but not for a subdomain of a subdomain (of a subdomain), as is the case with `service.namespace.svc.cluster.local`.

**Comment by 4censord** [comment]
oh, so for TLS certs it differs from how wildcards work for DNS. Because with dns, `*.cluster.local` would also resolve to `service.namespace.svc.cluster.local`
I did not know that.

**Comment by eyenx** [comment]
> > The TLS skip-verify is only needed cause Nutanix's Implementation of ETCD
> 
> Should we also reflect this issue in upstream Nutanix so it gets unborked at some point?

I'll discuss this with the customer tomorrow.

**Comment by eyenx** [comment]
This needs now a rebase

**Comment by in0rdr** [comment]
rebased, I think I almost need to bump to 1.3.0 now

**Comment by in0rdr** [comment]
@eyenx you only keep the latest artifacthub annotation right?

**Comment by eyenx** [comment]
yes, bump to 1.3.0 and just add the your change to artifacthub

**Pull request merged**